endpointmanager/pkg/endpointmanager/postgresql/fhirendpointmetadatastore_test.go

```diff
@@ -254,5 +256,90 @@ func Test_PersistFHIREndpointMetadata(t *testing.T) {
+	if err != nil {
+		t.Error(err)
+	}
+	cs, err := capabilityparser.NewCapabilityStatement(csJSON)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var endpointMetadata1 = &endpointmanager.FHIREndpointMetadata{
+		URL:               "example.com/FHIR/DSTU2/",
+		HTTPResponse:      200,
+		Errors:            "Example Error",
+		SMARTHTTPResponse: 0,
+		Availability:      1.0,
+		RequestedFhirVersion: "None"}
+
+	var endpointMetadata2 = &endpointmanager.FHIREndpointMetadata{
+		URL:               "other.example.com/FHIR/DSTU2/",
+		HTTPResponse:      404,
+		Errors:            "Example Error 2",
+		SMARTHTTPResponse: 0,
+		Availability:      0,
+		RequestedFhirVersion: "1.0.2"}
+
+	// endpointInfos
+	var endpointInfo1 = &endpointmanager.FHIREndpointInfo{
+		URL:                   "example.com/FHIR/DSTU2/",
+		TLSVersion:            "TLS 1.1",
+		MIMETypes:             []string{"application/json+fhir"},
+		CapabilityStatement:   cs,
+		RequestedFhirVersion:  "None",
+		CapabilityFhirVersion: "1.0.2",
+		SMARTResponse:         nil,
+		Metadata:              endpointMetadata1}
+
+	var endpointInfo2 = &endpointmanager.FHIREndpointInfo{
+		URL:                   "other.example.com/FHIR/DSTU2/",
+		TLSVersion:            "TLS 1.2",
+		RequestedFhirVersion:  "1.0.2",
+		CapabilityFhirVersion: "",
+		MIMETypes:             []string{"application/fhir+json"},
+		Metadata:              endpointMetadata2}
+
+	// add endpointMetadata
+
+	_, err = store.AddFHIREndpointMetadata(ctx, endpointMetadata1)
+	if err != nil {
+		t.Errorf("Error adding fhir endpointInfo: %s", err.Error())
+	}
+
+	_, err = store.AddFHIREndpointMetadata(ctx, endpointMetadata2)
+	if err != nil {
+		t.Errorf("Error adding fhir endpointInfo: %+v", err)
+	}
 
+	// check that endpoint availability was updated
+	var http_200_ct int
+	var http_all_ct int
+	var endpt_availability_ct int
+	query_str := "SELECT http_200_count, http_all_count from fhir_endpoints_availability WHERE url=$1 AND requested_fhir_version=$2;"
+	ct_availability_str := "SELECT COUNT(*) from fhir_endpoints_availability;"
+
+	err = store.DB.QueryRow(ct_availability_str).Scan(&endpt_availability_ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, endpt_availability_ct == 2, "endpoint availability should have 2 entries")
+	err = store.DB.QueryRow(query_str, endpointInfo1.URL, "None").Scan(&http_200_ct, &http_all_ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, http_all_ct == 1, "endpoint with requested version None should have http return count of 1")
+	th.Assert(t, http_200_ct == 1, "endpoint with requested version None should have http 200 return count of 1")
+	err = store.DB.QueryRow(query_str, endpointInfo2.URL, "1.0.2").Scan(&http_200_ct, &http_all_ct)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, http_all_ct == 1, "endpoint with requested version None should have http return count of 1")
+	th.Assert(t, http_200_ct == 0, "endpoint with requested version None should have http 200 return count of 0")
```

Comment:
Could add a third metadata object that has the same URL as one of the other two but a different RequestedFhirVersion to make sure it adds it's own row in the availability table and doesn't impact the availability currently there?

Reply:
Yes! Very good suggestion. Will add